agent/dns.go

```diff
@@ -1141,8 +1153,8 @@ func (d *DNSServer) lookupServiceNodes(cfg *dnsConfig, datacenter, service, tag
 }
 
 // serviceLookup is used to handle a service query
-func (d *DNSServer) serviceLookup(cfg *dnsConfig, network, datacenter, service, tag string, entMeta *structs.EnterpriseMeta, connect bool, req, resp *dns.Msg, maxRecursionLevel int) {
-	out, err := d.lookupServiceNodes(cfg, datacenter, service, tag, entMeta, connect, maxRecursionLevel)
+func (d *DNSServer) serviceLookup(cfg *dnsConfig, network, datacenter, service, tag string, entMeta *structs.EnterpriseMeta, connect, ingress bool, req, resp *dns.Msg, maxRecursionLevel int) {
```

Comment:
Tried out a refactor that does this, I could take it or leave it honestly. I wanted to refactor all of the different lookup functions to take a query struct but ended up thinking the struct would have to many extraneous fields that are only relevant to `nodes` or `prepared_query` or `service` lookups.

Pushed it so that people can take a look though!